carbon-feature-plugin/src/main/java/org/wso2/maven/p2/utils/ProductFileUtils.java

```
@@ -0,0 +1,159 @@
+    }
+
+    private static ProductConfig
+    validateProductConfigForVersionAndFeatureConfig(ProductConfig productConfig, MavenProject mavenProject)
+            throws MojoExecutionException {
+
+        if (productConfig.getVersion() == null || productConfig.getVersion().isEmpty()) {
+            productConfig.setVersion(ProductFileUtils.getDependencyVersion(mavenProject,
+                    P2Constants.ProductFileDefaults.Product.RUNTIME_FEATURE));
+        }
+        if (productConfig.getFeatureConfig() == null) {
+            productConfig.setFeatureConfig(ProductFileUtils.getDefaultFeatureConfig(mavenProject));
+        }
+
+        return productConfig;
+    }
+
+    private static List<Feature> getDefaultFeatureConfig(MavenProject mavenProject) throws MojoExecutionException {
+        List<Feature> features = new ArrayList<>();
+        // adding default features with versions
+        features.add(new Feature(P2Constants.ProductFileDefaults.Product.RUNTIME_FEATURE,
+                ProductFileUtils.getDependencyVersion(mavenProject,
+                        P2Constants.ProductFileDefaults.Product.RUNTIME_FEATURE)));
+        return features;
+    }
+
+    private static String getDependencyVersion(MavenProject mavenProject,
+                                               String dependency) throws MojoExecutionException {
+        if (mavenProject == null) {
+            throw new MojoExecutionException("Unable to read maven project for finding dependencies.");
+        } else {
+            String dependencyVersion = null;
+            for (Artifact artifact : mavenProject.getDependencyArtifacts()) {
+                if (artifact.getArtifactId() != null &&
+                        artifact.getArtifactId().contains(dependency)) {
+                    dependencyVersion = artifact.getBaseVersion();
+                    break;
+                }
+            }
+            return dependencyVersion;
```

If the dependencyVersion is null, then it should throw an error and the build should stop.

---

Fixed by commit 8be4f48bc1afacdcba5c889fc630c7e7140939a0.